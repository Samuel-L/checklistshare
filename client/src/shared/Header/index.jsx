import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';

const styles = {
  root: {
    flexGrow: 1,
  },
  flex: {
    flex: 1,
  },
};

const Header = props => (
  <AppBar position="static" className={props.classes.root}>
    <Toolbar>
      <Typography variant="title" color="inherit" className={props.classes.flex}>
        ChecklistShare
      </Typography>
      <Button color="inherit" component={Link} to="/">
        Home
      </Button>
    </Toolbar>
  </AppBar>
);

Header.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(Header);
