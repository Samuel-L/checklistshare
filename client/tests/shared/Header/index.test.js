import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';

import Header from '../../../src/shared/Header';

describe('shared: Header', () => {
  const classes = { root: '', flex: '' };
  const tree = renderer.create(<HashRouter><Header classes={classes} /></HashRouter>).toJSON();

  it('should render', () => {
    expect(tree).toMatchSnapshot(); 
  });
});
