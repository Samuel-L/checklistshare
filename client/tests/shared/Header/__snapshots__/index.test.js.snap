// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: Header should render 1`] = `
<header
  className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-3 MuiAppBar-positionStatic-7 MuiAppBar-colorPrimary-9 Header-root-1"
>
  <div
    className="MuiToolbar-root-38 MuiToolbar-gutters-39"
  >
    <h2
      className="MuiTypography-root-40 MuiTypography-title-46 MuiTypography-colorInherit-59 Header-flex-2"
    >
      ChecklistShare
    </h2>
    <a
      className="MuiButtonBase-root-79 MuiButton-root-64 MuiButton-colorInherit-68"
      href="#/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-65"
      >
        Home
      </span>
      <span
        className="MuiTouchRipple-root-82"
      />
    </a>
  </div>
</header>
`;
